huaweicloud/services/cdn/resource_huaweicloud_cdn_domain_rule.go

```diff
@@ -0,0 +1,948 @@
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API CDN GET /v1.0/cdn/configuration/domains/{domain_name}/rule
+// @API CDN PUT /v1.0/cdn/configuration/domains/{domain_name}/rules/full-update
+
+// Please apply for whitelist permission before using this resource.
+// The API used by this resource is an offline document and has no official website. So no documentation is provided yet.
+func ResourceCdnDomainRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCdnDomainRuleCreate,
+		ReadContext:   resourceCdnDomainRuleRead,
+		UpdateContext: resourceCdnDomainRuleUpdate,
+		DeleteContext: resourceCdnDomainRuleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCDNDomainRuleImportState,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"rules": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "Specifies a list of rules.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Specifies the rule name. The valid length is limit from `1` to `50`.",
+						},
+						"status": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Specifies the rule status. Valid values are **on** and **off**.",
+						},
+						"priority": {
+							Type:        schema.TypeInt,
+							Required:    true,
+							Description: `Specifies the rule priority. The valid value is limit from 1 to 100.`,
+						},
+						"conditions": ruleConditions(),
+						"actions":    ruleActions(),
+						"rule_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func ruleConditionsMatch() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"logic": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Specifies the logical operator. Valid values are **and** and **or**.",
+				},
+				"criteria": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsJSON,
+					DiffSuppressFunc: func(_, o, n string, _ *schema.ResourceData) bool {
+						return utils.JSONStringsEqual(o, n)
+					},
+					Description: "Specifies the match criteria list in JSON format.",
+				},
```

Comment:
```suggestion
				"criteria": {
					Type:      schema.TypeList,
					Required:  true,
					Elem:      &schema.Schema{
						Type: schema.TypeString,
						ValidateFunc: validation.StringIsJSON,
					}
					DiffSuppressFunc: ...
					Description: "Specifies the match criteria list in JSON format.",
				},
```
seems this parameter is a JSON string list and with type list schema definition.

Reply:
This field is a linked list structure, and it is more appropriate to define it as json as a whole.This opinion will not be modified.